### DISCUSSION TITLE
Need Help With Learning Nodejs 😥

### DISCUSSION THREAD

**Discussion by xxFREESHROUDxx** [community/community#67564] [Programming Help]
### Select Topic Area

Question

### Body

I am recently trying to learn Nodejs in depth. I have a basic knowledge in Node, how to create server and routes but I want to learn it in like a professional way like how backend developers make APIs and stuff in Nodejs by properly structuring the code directory and writing the codes. 

I would love to know which is the best way to learn Nodejs for beginners? I have a good knowledge in React and Javascript but I want to learn Node in a proper way. 

Please suggest some youtube channels or documentations or any website like freecodecamp where I can learn about Node as well as practically learn to code side by side.

**Comment by priyadarshss** [comment]
You can start with the official docs to get a rough idea. Traversy Media and Net Ninja on YouTube are gold. freeCodeCamp also offers solid interactive lessons. You can level up by understanding the MVC pattern and diving into Express.js. Build multiple projects and learn as you go. If you're ever stuck, ChatGPT is always there along with Stack overflow and reddit's Node community.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @xxFREESHROUDxx ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
